internal/services/desktopvirtualization/virtual_desktop_host_pool_data_source.go

```diff
@@ -0,0 +1,188 @@
+		Read: dataSourceVirtualDesktopHostPoolRead,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Read: pluginsdk.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"location": commonschema.LocationComputed(),
+
+			"resource_group_name": commonschema.ResourceGroupName(),
+
+			"type": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"load_balancer_type": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"friendly_name": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"description": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"validate_environment": {
+				Type:     pluginsdk.TypeBool,
+				Computed: true,
+			},
+
+			"custom_rdp_properties": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"personal_desktop_assignment_type": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"maximum_sessions_allowed": {
+				Type:     pluginsdk.TypeInt,
+				Computed: true,
+			},
+
+			"start_vm_on_connect": {
+				Type:     pluginsdk.TypeBool,
+				Computed: true,
+			},
+
+			"preferred_app_group_type": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"scheduled_agent_updates": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"enabled": {
+							Type:     pluginsdk.TypeBool,
+							Computed: true,
+						},
+
+						"timezone": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"use_session_host_timezone": {
+							Type:     pluginsdk.TypeBool,
+							Computed: true,
+						},
+
+						"schedule": {
+							Type:     pluginsdk.TypeList,
+							Computed: true,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"day_of_week": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+
+									"hour_of_day": {
+										Type:     pluginsdk.TypeInt,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"tags": commonschema.Tags(),
+		},
+	}
+}
+
+func dataSourceVirtualDesktopHostPoolRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DesktopVirtualization.HostPoolsClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id := hostpool.NewHostPoolID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
+
+	resp, err := client.Get(ctx, id)
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			log.Printf("[DEBUG] %s was not found - removing from state!", id)
+			d.SetId("")
+			return nil
```

Comment:
since this is a Data Source this should be returning an error when this isn't found, rather than marking it as gone:

```suggestion
			return fmt.Errorf("%s was not found", id)
```